addons/calendar/models/calendar_event.py

```diff
@@ -688,14 +688,23 @@ def write(self, values):
 
         return True
 
+    def _get_trigger_alarm_types(self):
+        return ['email']
+
     def _setup_alarms(self):
-        """ Trigger the cron in the future for every email reminder """
+        """ Schedule cron triggers for future events """
         cron = self.env.ref('calendar.ir_cron_scheduler_alarm')
+        alarm_manager = self.env['calendar.alarm_manager']
+        alarm_types = self._get_trigger_alarm_types()
+
         for event in self:
-            for alarm in (alarm for alarm in event.alarm_ids if alarm.alarm_type == 'email'):
-                cron._trigger(at=event.start-timedelta(minutes=alarm.duration_minutes))
+            for alarm in (alarm for alarm in event.alarm_ids if alarm.alarm_type in alarm_types):
+                at = event.start - timedelta(minutes=alarm.duration_minutes)
+                if at > cron.lastcall:
+                    # Don't trigger for past alarms, they would be skipped by design
+                    cron._trigger(at=at)
```

Comment:
We don't want to actively send reminder for any past alarm. Imagine should schedule at 8 o'clock a meeting at 16 o'clock with a 1h reminder. At 15h30 you decide to rename the meeting (or anything else), you don't want to resend an alarm. If you schedule at 8 a meeting at 9 with a 2h reminder the attendees receive an invitation email already so the reminder is not really needed.

Reply:
It is waaaaay to early to understand all that computation.

:coffee: